sys/net/application_layer/rdcli_simple/rdcli_simple_standalone.c

```diff
@@ -0,0 +1,54 @@
+/*
+ * Copyright (C) 2017 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     net_rdcli_simple
+ * @{
+ *
+ * @file
+ * @brief       Standalone extension for the simple RD registration client
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include "log.h"
+#include "thread.h"
+#include "xtimer.h"
+#include "net/rdcli_config.h"
+#include "net/rdcli_simple.h"
+
+#define STACKSIZE           (THREAD_STACKSIZE_DEFAULT)
+#define PRIO                (THREAD_PRIORITY_MAIN - 1)
+#define TNAME               "rdcli_simple"
+
+char stack[STACKSIZE];
+
+static void *reg_runner(void *arg)
+{
+    (void)arg;
+
+    /* wait some seconds to give the address configuration some time to settle */
+    xtimer_sleep(RDCLI_STARTUP_DELAY);
+
+    while (1) {
+        int res = rdcli_simple_register();
+        if (res < 0) {
+            LOG_ERROR("[rdcli_simple] error: unable to trigger registration\n");
+        }
+        xtimer_sleep(RDCLI_UPDATE_INTERVAL);
```

Comment:
I think it would be good to have a distinction here between a sleep interval when successfully registered and when triggering the registration failed. With this, if a registration fails, the next registration occurs at the exact moment the registration expires. With the default settings, if the node doesn't have connectivity within 3 seconds, it takes half a day before it registers.

I don't consider this blocking for now, it could always be done in a follow up PR.

Reply:
I missed that it sends confirmable messages, so this is already handled by the gcoap code :)

Reply:
:-)